### PR TITLE
Closes #570 Add title field to Cards paragraph type

### LINKED ISSUE #570: Add title field to Cards paragraph type
## Motivation
Allow title to be added before the Cards without the need for a separate Text paragraph type.

## Proposed Resolution
Still need to consider how to implement formatting and style options.

### PR DESCRIPTION
**Pull request by djcelaya**
## Description
A possible solution for issue #570. The View (reference) paragraph type has a plain text title field already. Copying that approach over to the Cards paragraph type. This solution does not account for formatting or styling needs but will allow site builders to add a simple title to their Cards paragraphs without having to create a second Text paragraph immediately above.

I have updated the config for the Cards paragraph type to include the AZ Title field much like the View (reference) paragraph type does. I have exported all the config files I believe I needed for this change into the config/install directory of the az_paragraphs_cards module. However, I suspect this will not be sufficient to update existing sites with this paragraph type already enabled.

## Related issues

#570 

### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
   - [ ] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
   - [ ] Adding experimental module
   - [ ] Update experimental module
- **Minor release changes**
   - [X] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Upgrade experimental module to stable
   - [ ] Enable existing module by default or database update
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- **Other or unknown**
   - [ ] Other or unknown

### Drupal core
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major or minor level update
- **Other or unknown**
   - [ ] Other or unknown

### Drupal contrib projects
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major level update
- **Other or unknown**
   - [ ] Other or unknown

## Checklist

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: modules/custom/az_paragraphs/az_paragraphs_cards/config/install/field.field.paragraph.az_cards.field_az_title.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+status: true
+dependencies:
+  config:
+    - field.storage.paragraph.field_az_title
+    - paragraphs.paragraphs_type.az_cards
+id: paragraph.az_cards.field_az_title
+field_name: field_az_title
+entity_type: paragraph
+bundle: az_cards
+label: Title
```
Comment (joeparsons) [inline modules/custom/az_paragraphs/az_paragraphs_cards/config/install/field.field.paragraph.az_cards.field_az_title.yml:11]: Suggestion from 2023-02-01 PR Meeting:
```suggestion
label: Card group title
```

**Comment by joeparsons** [comment]
Follow-up issues/tasks identified during 2022-02-01 PR meeting:
- Add new title field to card preview
- Update demo content to use title instead of separate text paragraph

**Review by trackleft — APPROVED** [review]

**Comment by trackleft** [comment]
Talked to @djcelaya offline, and we were in agreement that this PR is ready for review and approval, and merging.

**Review by trackleft — APPROVED** [review]

**Comment by joshuasosa** [comment]
It works but is kind of simplistic. I'd like to see the Title heading level and Title alignment be added here from the Text on Media paragraph.

This PR does reduce the need for yet another text paragraph, which is probably good. But I think the idea could go further: Title field for Photo Gallery and Accordion paragraphs.

**Comment by djcelaya** [comment]
Those are great suggestions about the Title heading level and Title alignment (from Text on Media paragraph), @joshuasosa.  I can probably get those added in next week if we can wait a bit on merging this.

I also agree with adding these 3 elements (title, title heading level, and title alignment) to the Photo Gallery and Accordion paragraphs. Should I include those on this PR or is that a separate item?

**Comment by akslay** [comment]
Adding in those changes might actually be blocked by pending group discussion/decisions from this PR:
https://github.com/az-digital/az_quickstart/pull/2041

The same changes are supposed to happen there, but the group hasn't come to a consensus yet on how much flexibility they do/don't want to allow for card titles. We're basically just waiting for time in a workshop to talk about it.

**Review by HenryzGreenberg — APPROVED** [review]

**Review by joeparsons — APPROVED** [review]

**Comment by joshuasosa** [comment]
> I also agree with adding these 3 elements (title, title heading level, and title alignment) to the Photo Gallery and Accordion paragraphs. Should I include those on this PR or is that a separate item?

I added issue #2205 to include the other paragraph types.

**Pull request merged**